repair/row_level.cc

```diff
@@ -2302,20 +2328,25 @@ future<> repair_service::init_ms_handlers() {
     });
     ms.register_repair_row_level_start([this] (const rpc::client_info& cinfo, uint32_t repair_meta_id, sstring ks_name,
             sstring cf_name, dht::token_range range, row_level_diff_detect_algorithm algo, uint64_t max_row_buf_size, uint64_t seed,
-            unsigned remote_shard, unsigned remote_shard_count, unsigned remote_ignore_msb, sstring remote_partitioner_name, table_schema_version schema_version, rpc::optional<streaming::stream_reason> reason) {
+            unsigned remote_shard, unsigned remote_shard_count, unsigned remote_ignore_msb, sstring remote_partitioner_name, table_schema_version schema_version,
+            rpc::optional<streaming::stream_reason> reason, rpc::optional<gc_clock::time_point> compaction_time) {
         auto src_cpu_id = cinfo.retrieve_auxiliary<uint32_t>("src_cpu_id");
         auto from = cinfo.retrieve_auxiliary<gms::inet_address>("baddr");
         return container().invoke_on(src_cpu_id % smp::count, [from, src_cpu_id, repair_meta_id, ks_name, cf_name,
-                range, algo, max_row_buf_size, seed, remote_shard, remote_shard_count, remote_ignore_msb, schema_version, reason, this] (repair_service& local_repair) mutable {
+                range, algo, max_row_buf_size, seed, remote_shard, remote_shard_count, remote_ignore_msb, schema_version, reason, compaction_time, this] (repair_service& local_repair) mutable {
             if (!local_repair._sys_dist_ks.local_is_initialized() || !local_repair._view_update_generator.local_is_initialized()) {
                 return make_exception_future<repair_row_level_start_response>(std::runtime_error(format("Node {} is not fully initialized for repair, try again later",
                         utils::fb_utilities::get_broadcast_address())));
             }
             streaming::stream_reason r = reason ? *reason : streaming::stream_reason::repair;
+            std::optional<gc_clock::time_point> ct;
+            if (compaction_time) {
+                ct = *compaction_time;
+            }
```

Comment:
Why not use optional\<> throughout? "0 is a special case" is just more more thing to keep track of and possibly forget.

Reply:
`compaction_time` is an `rpc::optional<>`. I cannot use `std::optional<>` in RPC, because RPC uses `rpc::optional<>` to denote parameter which might or might not be sent.

Reply:
rpc::optional`<std::optional`<...>>?

Reply:
Actually, in v2 I move away from `std::optional<gc_clock::time_point>` completely and just use `gc_clock::time_point`.